crowbar_framework/lib/crowbar/error/upgrade_status.rb

```diff
@@ -29,8 +29,9 @@ def initialize(step_name = "")
     end
 
     class StartStepOrderError < StandardError
-      def initialize(step_name = "")
-        super("Start of step #{step_name} requested in the wrong order")
+      def initialize(step_name = "", correct_step = "")
+        super("Start of step '#{step_name}' requested in the wrong order. " \
+        "Correct next step is '#{correct_step}'.")
```

Comment:
> Well, the concern is the same when step_name is empty, right?

`Start of step requested in the wrong order.` is still a correct sentence.
`Correct next step is .` is obviously incomplete.

why not just:
```ruby
      def initialize(step_name = "", correct_step = "")
        msg = "Start of step '#{step_name}' requested in the wrong order. "
        msg += "Correct next step is '#{correct_step}'." unless correct_step.empty?
        super(msg)
      end
```